test/integration/TokenRejectIntegrationTest.js

```diff
@@ -0,0 +1,834 @@
+                .toInt();
+            expect(tokenBalanceReceiver).to.equal(0);
+        });
+
+        it("should return spender allowance to 0 after owner rejects NFT", async function () {
+            this.timeout(120000);
+
+            // create spender account
+            const spenderAccountPrivateKey = PrivateKey.generateED25519();
+            const spenderAccountTx = await new AccountCreateTransaction()
+                .setMaxAutomaticTokenAssociations(-1)
+                .setInitialBalance(new Hbar(10))
+                .setKey(spenderAccountPrivateKey)
+                .execute(env.client);
+
+            const { accountId: spenderAccountId } =
+                await spenderAccountTx.getReceipt(env.client);
+
+            // transfer nft to receiver
+            const nftId = new NftId(tokenId, 1);
+            await (
+                await new TransferTransaction()
+                    .addNftTransfer(nftId, env.operatorId, receiverId)
+                    .execute(env.client)
+            ).getReceipt(env.client);
+
+            // approve nft allowance
+            await (
+                await (
+                    await new AccountAllowanceApproveTransaction()
+                        .approveTokenNftAllowance(
+                            nftId,
+                            receiverId,
+                            spenderAccountId,
+                        )
+                        .freezeWith(env.client)
+                        .sign(receiverPrivateKey)
+                ).execute(env.client)
+            ).getReceipt(env.client);
+
+            // reject nft
+            await (
+                await (
+                    await new TokenRejectTransaction()
+                        .addNftId(nftId)
+                        .setOwnerId(receiverId)
+                        .freezeWith(env.client)
+                        .sign(receiverPrivateKey)
+                ).execute(env.client)
+            ).getReceipt(env.client);
+
+            // transfer nft from receiver to spender using allowance
+            try {
+                const transactionId = TransactionId.generate(spenderAccountId);
+                await (
+                    await (
+                        await new TransferTransaction()
+                            .addApprovedNftTransfer(
+                                nftId,
+                                receiverId,
+                                spenderAccountId,
+                            )
+                            .setTransactionId(transactionId)
+                            .freezeWith(env.client)
+                            .sign(spenderAccountPrivateKey)
+                    ).execute(env.client)
+                ).getReceipt(env.client);
+            } catch (err) {
+                expect(err.message).to.include(
+                    "SPENDER_DOES_NOT_HAVE_ALLOWANCE",
+                );
+            }
+        });
+
+        describe("should revert", function () {
```

Comment:
nit: `revert` is misleading here as the term stands for another thing. But not that big of a deal.